### PR TITLE
Adds basic SKAdNetwork integration to report conversion events

### PR DESCRIPTION
**Pull request by darinwf**
Introduces `ConversionLogger` class to handle the reporting logic. Once we are on Xcode 13.3, we can adopt the new iOS 15.4 `updatePostbackConversionValue` method. For now just sticking to the original methods.

I haven't done any testing of this yet. Hence, just a draft. Would love any feedback on this approach though, and especially on where I've inserted the various events.

I think it will be a separate PR to add the `Info.plist` key.

### How was this tested?
- [x] I tested this manually

### CODE REVIEW

File: Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserIntestitialOnboardingView.swift [hunk 1]
```diff
@@ -104,6 +104,7 @@ struct DefaultBrowserInterstitialOnboardingView: View {
                         Defaults[.lastDefaultBrowserPromptInteraction] =
                             LogConfig.Interaction.DefaultBrowserOnboardingInterstitialOpen.rawValue
                     }
+                    ConversionLogger.log(event: .visitedDefaultBrowserSettings)
```
[the remark this answers is not in the document]
Comment (darinwf) [reply Client/Frontend/DefaultBrowserOnboarding/DefaultBrowserIntestitialOnboardingView.swift:107]: Thanks, I was wondering about this too. I was a bit unsure if we'd want the conversion to be specific to the default browser interstitial or to just apply whenever the user clicks out to visit OS settings. Thinking about it some more, I agree we should just count any click out. Doesn't have to be specific to the first run promo.

**Comment by darinwf** [comment]
Another thought I had was to maybe add some gaps between events 0, 1 and 2. Maybe they should be 0, 5 and 10 instead. That way if we ever have other intermediate events we want to log, we could invent those without needing to redefine the existing events.

**Comment by chung-lai** [comment]
> Another thought I had was to maybe add some gaps between events 0, 1 and 2. Maybe they should be 0, 5 and 10 instead. That way if we ever have other intermediate events we want to log, we could invent those without needing to redefine the existing events.

That makes sense especially the logic right now is pretty straightforward as the larger number will always replace the smaller ones. I just don't know how much the gap we should be setting with it being a 6 bit number. May be 0, 5, 10 are good :D

**Comment by darinwf** [comment]
I went with 0, 10, 20 :-D

**Comment by chung-lai** [comment]
> I went with 0, 10, 20 :-D

Sounds good to start with. LGTM, will take another look once the PR is up for review

**Review by chung-lai — APPROVED** [review]
LGTM!

**Comment by chung-lai** [comment]
Do we want to cherry pick this into release?

**Comment by darinwf** [comment]
> Do we want to cherry pick this into release?

Probably a good idea. I'm checking on if this will be needed this week. I may also need to update this to work with your change to first run to ensure that initial conversion event is still reported.

**Comment by darinwf** [comment]
@chung-lai - You should land your PR first, and then I'll work on updating this one.

**Review by chung-lai — APPROVED** [review]

**Comment by darinwf** [comment]
Note to @chassu - This PR depends on https://github.com/neevaco/neeva-ios/pull/3193

**Pull request merged**